src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -137,21 +209,38 @@ public BinaryData(string data)
         /// <returns>A value representing the UTF-8 encoding of <paramref name="data"/>.</returns>
         public static BinaryData FromString(string data) => new BinaryData(data);
 
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance from a string by converting
+        /// the string to bytes using the UTF-8 encoding
+        /// and sets <see cref="MediaType"/> to <see pref="mediaType"/> value.
+        /// </summary>
+        /// <param name="data">The string data.</param>
+        /// <param name="mediaType">MIME type of this data, e.g. <see cref="MediaTypeNames.Text.Plain"/>.</param>
+        /// <returns>A value representing the UTF-8 encoding of <paramref name="data"/>.</returns>
+        /// <seealso cref="MediaTypeNames"/>
+        public static BinaryData FromString(string data, string? mediaType)
+            => new BinaryData(data, mediaType);
+
         /// <summary>
         /// Creates a <see cref="BinaryData"/> instance from the specified stream.
         /// The stream is not disposed by this method.
         /// </summary>
         /// <param name="stream">Stream containing the data.</param>
         /// <returns>A value representing all of the data remaining in <paramref name="stream"/>.</returns>
         public static BinaryData FromStream(Stream stream)
-        {
-            if (stream is null)
-            {
-                throw new ArgumentNullException(nameof(stream));
-            }
+            => FromStreamAsync(stream, async: false).GetAwaiter().GetResult();
```

Comment:
Why are we not calling stream sync APIs here and instead are doing sync over async?

Reply:
ah, I see it's the async:false trick. Never mind then